mrmustard/utils/training.py

```diff
@@ -74,7 +77,9 @@ def minimize(
                     self.opt_history.append(cost)
                     bar.step(math.asnumpy(cost))
         except KeyboardInterrupt:  # graceful exit
-            return
+            raise self.OptimizerInterruptedError(
+                "Stopping optimizer execution...", self.log
```

Comment:
Perhaps rewording this slightly since it's now an exception message instead of a printed statement. For example, "Optimizer execution halted due to keyboard interruption", or something similar.

Reply:
Agree!